### PR TITLE
Documentation: engine.py

### PR DESCRIPTION
**Pull request by nahuakang**
This PR addresses [Issue #26](https://github.com/TwentyBN/20bn-realtimenet/issues/26). It adds some inline documentation and type hints to `engine.py`.

### CODE REVIEW

File: realtimenet/engine.py [hunk 1]
```diff
@@ -67,7 +100,27 @@ def run(self):
...
+        of numpy.ndarray.
+
+        For an inference engine running 1 classifier, the output is a numpy.ndarray
+        of shape (row, col), where `col` represents the number of labels that the
+        classifier predicts.
+
+        For an inference engine running more than 1 classifier, the output is a list
+        of N numpy.ndarray, where `N` represents the number of classifiers that the
+        inference engine runs. Each of the numpy.ndarray is of shape (row, col),
+        where `col` represents the number of labels that the classifier predicts.
```
Comment (guillaumebrg) [inline realtimenet/engine.py:116]: I just have a problem with using the words "classifier" / "number of labels that the classifier predicts" because this assumes a classification task. Our MET predictor, for instance, is solving a regression task and doesn't have "labels". 

I think we can rephrase this using more generic terms. How about something like this?

```
For an inference engine running a neural network producing a single output, the returned 
object is a numpy.ndarray of shape (T, C), where `C` represents the number of output 
channels while `T` represents the number of time steps and depends on the length 
of the provided input clip.

For an inference engine running a multi-output neural network, the returned object is a 
list of numpy.ndarray, one for each output.
```

File: realtimenet/engine.py [hunk 2]
```diff
@@ -84,7 +137,28 @@ def infer(self, clip):
...
+    results_display: DisplayResults,
+    path_out: Optional[str]
+):
+    """
+    Start the video stream and the inference engine, process and display
+    the prediction from the neural network.
+
+    :param inference_engine:
+        An instance of InferenceEngine for neural network inferencing.
+    :param framegrabber:
```
Comment (guillaumebrg) [inline realtimenet/engine.py:153]: Future PR: rename `framegrabber` to `video_stream`

**Review by guillaumebrg — COMMENTED** [review]
Thanks @nahuakang!

I've made a few comments

**Comment by guillaumebrg** [comment]
Cc #26

File: realtimenet/engine.py [hunk 3]
```diff
@@ -1,14 +1,32 @@
...
 
 
 class InferenceEngine(Thread):
-
-    def __init__(self, net, use_gpu=False):
+    """
+    InferenceEngine takes in a neural network and uses it to output predictions
+    either using the local machine's CPU or GPU.
+    """
+    def __init__(self, net: Pipe, use_gpu: bool = False):
```
[the remark this answers is not in the document]
Comment (nahuakang) [reply realtimenet/engine.py:23]: So I've renamed the type to `nn.Module` since all the current feature extractors and converters inherit from `nn.Module` :smile:

File: realtimenet/engine.py (hunk 1 above)
Comment (nahuakang) [reply realtimenet/engine.py:116]: Great! I edited it a bit as well :)

File: realtimenet/engine.py (hunk 3 above)
[the remark this answers is not in the document]
Comment (guillaumebrg) [reply realtimenet/engine.py]: Makes sense!

**Review by guillaumebrg — APPROVED** [review]
LGTM! :+1:
